### PR TITLE
Remove upload data #137

### PR DESCRIPTION
**Pull request by olenagerasimova**
For #137 added `com.artipie.docker.Upload#delete`, implementation, test and todo to continue.

### CODE REVIEW

File: src/main/java/com/artipie/docker/http/UploadEntity.java [hunk 1]
```diff
@@ -159,26 +166,30 @@ public Response response(
...
+                                        ),
+                                        new Header("Content-Length", "0"),
+                                        new DigestHeader(digest)
+                                    )
                                 );
                             } else {
-                                response = new RsWithStatus(RsStatus.BAD_REQUEST);
+                                response = CompletableFuture.completedStage(
+                                    new RsWithStatus(RsStatus.BAD_REQUEST)
+                                );
```
Comment (olegmoz) [inline src/main/java/com/artipie/docker/http/UploadEntity.java:192]: @olenagerasimova this code is not covered with tests in this PR and I believe it requires coverage. I'd suggest adding slice level tests to this PR or creating new PR with changes to `UploadEntity` class and tests for it.

File: src/test/java/com/artipie/docker/asto/AstoUploadTest.java [hunk 2]
```diff
@@ -135,4 +143,16 @@ void shouldAppendedSecondChunkIfFirstOneFailed() throws Exception {
             new IsEqual<>(chunk)
         );
     }
+
+    @Test
+    void shouldRemoveUploadedFiles() throws ExecutionException, InterruptedException {
+        final byte[] chunk = "some bytes".getBytes();
+        this.upload.append(Flowable.just(ByteBuffer.wrap(chunk))).toCompletableFuture().get();
+        this.upload.content().toCompletableFuture().get();
```
Comment (olegmoz) [inline src/test/java/com/artipie/docker/asto/AstoUploadTest.java:151]: @olenagerasimova do not see any reason for reading content in this test, could you explain please?

**Comment by olenagerasimova** [comment]
@olegmoz thanks, corrected, please, take a look

**Comment by olegmoz** [comment]
@olenagerasimova thank you, everything looks great

**Pull request merged**